tools/generator/test/DisambiguateTargetsTests.swift

```diff
@@ -23,12 +23,23 @@ final class DisambiguateTargetsTests: XCTestCase {
                 label: "//a:B",
                 product: .init(type: .staticLibrary, name: "B", path: "")
             ),
+            // The following targets only differ by case
+            "C 1": Target.mock(
+                label: "//c:A",
+                product: .init(type: .staticLibrary, name: "A", path: "")
+            ),
+            "C 2": Target.mock(
+                label: "//c:a",
+                product: .init(type: .application, name: "a", path: "")
+            ),
```

Comment:
Can you add an example to `test_productType` to show that it does this for non-label based results as well?